tests/unittests/sources/test___init__.py

```diff
@@ -4,42 +4,44 @@
 from cloudinit.sources import DataSourceOpenStack as ds
 from tests.unittests.helpers import mock
 
+openstack_ds_name = ds.DataSourceOpenStack.dsname.lower()
+
 
 @pytest.mark.parametrize(
-    "m_cmdline",
+    "m_cmdline, expected_ds",
     (
         # test ci.ds=
-        "aosiejfoij ci.ds=OpenStack ",
-        "ci.ds=OpenStack",
-        "aosiejfoij ci.ds=OpenStack blah",
-        "aosiejfoij ci.ds=OpenStack faljskebflk",
-        "ci.ds=OpenStack;",
-        "ci.ds=openstack;",
+        ("aosiejfoij ci.ds=OpenStack ", openstack_ds_name),
+        ("ci.ds=OpenStack", openstack_ds_name),
+        ("aosiejfoij ci.ds=OpenStack blah", openstack_ds_name),
+        ("aosiejfoij ci.ds=OpenStack faljskebflk", openstack_ds_name),
+        ("ci.ds=OpenStack;", openstack_ds_name),
+        ("ci.ds=openstack;", openstack_ds_name),
         # test ci.datasource=
-        "aosiejfoij ci.datasource=OpenStack ",
-        "ci.datasource=OpenStack",
-        "aosiejfoij ci.datasource=OpenStack blah",
-        "aosiejfoij ci.datasource=OpenStack faljskebflk",
-        "ci.datasource=OpenStack;",
-        "ci.datasource=openstack;",
+        ("aosiejfoij ci.datasource=OpenStack ", openstack_ds_name),
+        ("ci.datasource=OpenStack", openstack_ds_name),
+        ("aosiejfoij ci.datasource=OpenStack blah", openstack_ds_name),
+        ("aosiejfoij ci.datasource=OpenStack faljskebflk", openstack_ds_name),
+        ("ci.datasource=OpenStack;", openstack_ds_name),
+        ("ci.datasource=openstack;", openstack_ds_name),
         # weird whitespace
-        "ci.datasource=OpenStack\n",
-        "ci.datasource=OpenStack\t",
-        "ci.datasource=OpenStack\r",
-        "ci.datasource=OpenStack\v",
-        "ci.ds=OpenStack\n",
-        "ci.ds=OpenStack\t",
-        "ci.ds=OpenStack\r",
-        "ci.ds=OpenStack\v",
+        ("ci.datasource=OpenStack\n", openstack_ds_name),
+        ("ci.datasource=OpenStack\t", openstack_ds_name),
+        ("ci.datasource=OpenStack\r", openstack_ds_name),
+        ("ci.datasource=OpenStack\v", openstack_ds_name),
+        ("ci.ds=OpenStack\n", openstack_ds_name),
+        ("ci.ds=OpenStack\t", openstack_ds_name),
+        ("ci.ds=OpenStack\r", openstack_ds_name),
+        ("ci.ds=OpenStack\v", openstack_ds_name),
+        ("ci.ds=nocloud-net\v", "nocloud-net"),
```

Comment:
Should we add a nocloud (without the -net) here too?

Reply:
done.